miden-tx/src/compiler/tests.rs

```diff
@@ -157,15 +158,19 @@ fn mock_input_notes(tx_compiler: &mut TransactionCompiler, target_account: Accou
     const SERIAL_NUM_1: Word = [Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)];
     let vault =
         NoteAssets::new(vec![fungible_asset_1, fungible_asset_2, fungible_asset_3]).unwrap();
-    let metadata = NoteMetadata::new(sender, NoteType::Public, 0.into(), ZERO).unwrap();
+    let metadata =
+        NoteMetadata::new(sender, NoteType::Public, 0.into(), NoteExecutionHint::always(), ZERO)
+            .unwrap();
     let inputs = NoteInputs::new(vec![Felt::new(1)]).unwrap();
     let recipient = NoteRecipient::new(SERIAL_NUM_1, note_script.clone(), inputs);
     let note_1 = Note::new(vault, metadata, recipient);
 
     const SERIAL_NUM_2: Word = [Felt::new(5), Felt::new(6), Felt::new(7), Felt::new(8)];
     let vault =
         NoteAssets::new(vec![fungible_asset_1, fungible_asset_2, fungible_asset_3]).unwrap();
-    let metadata = NoteMetadata::new(sender, NoteType::Public, 0.into(), ZERO).unwrap();
+    let metadata =
+        NoteMetadata::new(sender, NoteType::Public, 0.into(), NoteExecutionHint::always(), ZERO)
+            .unwrap();
```

Comment:
I noticed that sometimes we use `NoteExecutionHint::always()` and sometimes `NoteExecutionHint::Always`. Any reason to do one vs. another?

Reply:
Yeah, there is no difference. I wanted to provide constructors for the more involved hints and ended up adding them for all variants. I can remove the constructors for the simple variants if preferred.

Reply:
I think it's fine to keep it in - but maybe I'd use them consistently (e.g., always use a constructor). A very minor thing though - so, feel free to ignore.